dap_aria_mapping/pipeline/taxonomy_validation/make_labels.py

```diff
@@ -0,0 +1,155 @@
+from dap_aria_mapping.utils.validation import *
+
+
+def save_labels(taxonomy_class: str, label_type: str, level: int, labels: Dict) -> None:
+    """Save the labels for a given taxonomy level.
+
+    Args:
+        taxonomy_class (str): A string representing the taxonomy class.
+            Can be 'cooccur', 'semantic' or 'semantic_kmeans'.
+        label_type (str): A string representing the labels type.
+        level (int): The level of the taxonomy to use.
+        labels (Dict): A dictionary of the labels.
+    """
+    title = "outputs/topic_labels/labels_class_{}_labtype_{}_level_{}.pkl".format(
+        taxonomy_class, label_type, str(level)
+    )
+    s3 = boto3.client("s3")
+    s3.put_object(
+        Body=pickle.dumps(labels),
+        Bucket=BUCKET_NAME,
+        Key=title,
+    )
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(
+        description="Make histograms of a given taxonomy level.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--taxonomy_class",
+        help="The type of taxonomy to use. Can be 'cooccur', 'semantic' or 'semantic_kmeans'.",
+        default="cooccur",
+    )
+
+    parser.add_argument(
+        "--label_type",
+        nargs="+",
+        help="The type of labels to use. Can be 'entity', 'journal', or both.",
+        default="entity",
+    )
+
+    parser.add_argument(
+        "--n_top",
+        type=int,
+        help="The number of top elements to show. Defaults to 3.",
+        default=3,
+    )
+
+    parser.add_argument(
+        "--levels", nargs="+", help="The levels of the taxonomy to use.", required=True
+    )
+
+    parser.add_argument(
+        "--n_articles",
+        type=int,
+        help="The number of articles to use. If -1, all articles are used.",
+    )
+
+    parser.add_argument("--save", help="Whether to save the plot.", action="store_true")
+
+    args = parser.parse_args()
+
+    logger.info("Loading data - taxonomy")
+    if args.taxonomy_class == "cooccur":
+        taxonomy = get_cooccurrence_taxonomy()
+    elif args.taxonomy_class == "semantic":
+        taxonomy = pipe(
+            "centroids",
+            get_semantic_taxonomy,
+            lambda df: df.rename(columns={"tag": "Entity"})
+            if "tag" in df.columns
+            else df,
+            lambda df: df.set_index("Entity"),
+            lambda df: df.rename(
+                columns={
+                    k: "Level_{}".format(str(v + 1)) for v, k in enumerate(df.columns)
+                }
+            ),
+        )
+    elif args.taxonomy_class == "semantic_kmeans":
+        taxonomy = pipe(
+            "kmeans_strict_imb",
+            get_semantic_taxonomy,
+            lambda df: df.rename(columns={"tag": "Entity"})
+            if "tag" in df.columns
+            else df,
+            lambda df: df.set_index("Entity"),
+            lambda df: df.rename(
```

Comment:
These are included here because I zero-indexed my levels. I'm changing it in branch 9_, so this should eventually be reverted.